src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -581,13 +638,13 @@ def engine_inputs_for_prefill(
             yield engine_inputs
 
     @property
-    def has_cache(self) -> bool:
+    def cache_support_enabled(self) -> bool:
         """
         Returns whether the ran model has kv cache or not
 
         :return: True if the model has kv cache, False otherwise
         """
-        return self.multitoken_engine.kv_cache_enabled
```

Comment:
store as private variable that is initialized to `None` in `__init__` and filled out on first call here, referenced after